CRM/Contribute/BAO/Contribution.php

```diff
@@ -879,6 +879,34 @@ public static function &exportableFields($checkPermission = TRUE) {
     return self::$_exportableFields;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function addSelectWhereClause() {
+    $whereClauses = parent::addSelectWhereClause();
+    $originalWhereClauses = $whereClauses;
+    if ($whereClauses !== $originalWhereClauses) {
```

Comment:
@eileenmcnaughton I'm confused about what this code is doing. It assigns a variable to another variable, and then immediately checks if they are equal.

Reply:
@colemanw did you read the comment below? Basically we have an issue where we have financial acls in the hook AND the extension & we don't want both to apply. Originally the only places that called the hook DIDN'T apply the financial acls & vice versa.

Long term we want to only use the extension or only call the lines below this return. Joe doesn't yet know that he agrees with me on which one :-)

Reply:
I did read the comment but the code still doesn't make logical sense to me. It's basically saying:

    function checkIfFooEqualsFoo() {
      $foo = rand();
      $bar = $foo;
      if ($bar !== $foo) {
        // this will never happen. Why are we checking for this?

Reply:
@colemanw it WOULD happen if the parent 's call to the financialACLHook hit the financialaclreport extension

Reply:
Sorry I still don't see how. Even if the `rand()` function in my example were to call a hook, it's still assigning the output of that hook to two variables which will always be exactly the same no matter what, and then comparing them to see if they are exactly the same (they are).

Reply:
ah - I mis-ordered when I brought across - I see now - fixed

Reply:
Ok now the first line of the function seems to be trying to access an undefined variable `$whereClauses`

Reply:
this time?

Reply:
Now it makes sense :)

Reply:
thanks for persevering with me!